game/scripts/vscripts/modules/hero_selection/util.lua

```diff
@@ -113,14 +113,21 @@ function TransformUnitClass(unit, classTable, skipAbilityRemap)
 			unit:SetBaseDamageMax(value)
 		elseif key == "AttackRate" then
 			unit:SetBaseAttackTime(value)
+			unit:SetNetworkableEntityInfo("AttackRate", value)
 		elseif key == "AttackAcquisitionRange" then
 			unit:SetAcquisitionRange(value)
 		elseif key == "ProjectileModel" then
 			unit:SetRangedProjectileName(value)
 		elseif key == "AttributePrimary" then
-			Timers:CreateTimer(0.1, function()
+			Timers:CreateTimer(1, function()
 				unit:SetPrimaryAttribute(_G[value])
-			end)
+				unit:CalculateStatBonus()
+				local illusionSource = unit.illusionSource
+				if illusionSource then
+					unit:SetHealth(illusionSource:GetHealth())
+					unit:SetMana(illusionSource:GetMana())
```

Comment:
Is there a way that mana won't be adjusted with stat recalculation?

Reply:
if you don't go to/from an int base